### PR TITLE
Travis setup

### PR DESCRIPTION
**Pull request by mloubout**
Attempt to go back to line by line TRAVIS to catch all errors.

- currently setup with failing adjoint test to verify it does what it should

- Yes the comments are removed. TRAVIS refuses to recognize the yml file as valid with the comments in.

### CODE REVIEW

**Comment by mloubout** [comment]
Looks like we're back on track. Log looks as before my mess up.

**Comment by mloubout** [comment]
@fablup so this is finally working as expected.

It is failing as designed too to verify TRAVIS works correctly. If you're happy with it I'll put back the correct test and should be goood to go.

File: .travis.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ matrix:
           packages:
             - gcc-5
             - g++-5
-      env: DEVITO_ARCH=gcc-5 DEVITO_OPENMP=0 INSTALL_TYPE=conda
+      env: DEVITO_ARCH=gcc-5 DEVITO_OPENMP=0 SEISMIC=examples
```
Comment (FabioLuporini) [inline .travis.yml]: can you leave INSTALL_TYPE=conda ? it'd make things easier to understand

File: .travis.yml [hunk 2]
```diff
@@ -99,36 +99,58 @@ before_script:
...
+  - if [[ $SEISMIC == 'examples' ]]; then
+      DEVITO_BACKEND=foreign py.test -vs tests/test_operator.py -k TestForeign;
+    fi
+  - if [[ $SEISMIC == 'examples' ]]; then
+      python examples/seismic/benchmark.py test -P tti -so 4 -a -d 20 20 20 -n 5;
+    fi
+  - if [[ $SEISMIC == 'examples' ]]; then
+      python examples/seismic/benchmark.py test -P acoustic -a;
+    fi
+  - if [[ $SEISMIC == 'examples' ]]; then
```
Comment (FabioLuporini) [inline .travis.yml]: do the examples fail if the ``if-then-else`` is not inline?

File: .travis.yml [hunk 3]
```diff
@@ -99,36 +99,58 @@ before_script:
...
+  # Run tests with pip
+  - if [[ $INSTALL_TYPE == 'pip_setup' ]]; then python setup.py test; fi
+  # Run test with py.tests
+  - if [[ $INSTALL_TYPE != 'pip_setup' ]]; then py.test -vs --cov devito tests/; fi
+  # Run remaining specialized examples and tests
+  # Additional seismic operator tests
+  - if [[ $SEISMIC == 'examples' ]]; then
+      DEVITO_BACKEND=foreign py.test -vs tests/test_operator.py -k TestForeign;
+    fi
+  - if [[ $SEISMIC == 'examples' ]]; then
```
Comment (FabioLuporini) [inline .travis.yml]: I prefer inlined ``if-then-else`` in this file, when possible

**Comment by tmbgreaves** [comment]
Fairly confident I can tidy that up in Jenkins, yes. Having spent the last few days neatening a Jenkinsfile for OK I have a bunch of ideas to simplify/neaten the devito Jenkinsfile.

**Comment by FabioLuporini** [comment]
> Fairly confident I can tidy that up in Jenkins, yes. Having spent the last few days neatening a Jenkinsfile for OK I have a bunch of ideas to simplify/neaten the devito Jenkinsfile.

Awesome!

File: .travis.yml (hunk 1 above)
Comment (mloubout) [reply .travis.yml]: Yes

File: .travis.yml (hunk 2 above)
Comment (mloubout) [reply .travis.yml]: no it's fine onliner too

**Comment by mloubout** [comment]
>The proliferation of if-then-else really starts scaring me.

Should technically only need one but we want to run the examples each individually

> SEISMIC

true, will rename, it `RUN_EXAMPLES` fine?

**Comment by FabioLuporini** [comment]
> true, will rename, it RUN_EXAMPLES fine?

OK. However, if ``RUN_EXAMPLES == True`` **if and only if** ``INSTALL_TYPE == conda``, then I wonder why we don't just use ``INSTALL_TYPE`` ?

**Comment by mloubout** [comment]
>if and only if INSTALL_TYPE == conda

No also `False` for YASK backend

**Comment by FabioLuporini** [comment]
I have got no idea why Codecov is complaining that much

**Comment by mloubout** [comment]
I have honestly no idea. Can't figure out why it has such a huge difference.

**Comment by FabioLuporini** [comment]
after quite in-depth inspection, I still can't spot anything odd with `codecov`. Merging

**Comment by FabioLuporini** [comment]
Merged.

**Pull request merged**